packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useUpdateBalance.ts

```diff
@@ -26,19 +25,20 @@ export function useUpdateBalance(chainId: ChainId) {
             balance: '0',
         })
         return
-    }, [currentAccount])
+    }, [account])
 
     useEffect(() => {
-        if (!currentAccount) return
-        const value = inputToken?.schema === SchemaType.Native ? balance.value : '0'
-        dispatchTradeStore({
-            type: AllProviderTradeActionType.UPDATE_INPUT_TOKEN_BALANCE,
-            balance: value || '0',
-        })
-    }, [currentAccount, inputToken?.schema, balance.value])
+        if (!account) return
+        if (inputToken?.schema === SchemaType.Native) {
```

Comment:
CC @UncleBill This is the critical point to resolving the issue. Any problem do you find here?